code/implementation/host-stats/src/main/java/io/cattle/platform/host/stats/api/StatsLinkHandler.java

```diff
@@ -0,0 +1,96 @@
+package io.cattle.platform.host.stats.api;
+
+import io.cattle.platform.api.link.LinkHandler;
+import io.cattle.platform.archaius.util.ArchaiusUtil;
+import io.cattle.platform.core.constants.HostConstants;
+import io.cattle.platform.core.constants.InstanceConstants;
+import io.cattle.platform.core.model.Host;
+import io.cattle.platform.core.model.Instance;
+import io.cattle.platform.docker.util.DockerUtils;
+import io.cattle.platform.host.api.HostApiUtils;
+import io.cattle.platform.host.model.HostApiAccess;
+import io.cattle.platform.host.service.HostApiService;
+import io.cattle.platform.host.stats.utils.StatsConstants;
+import io.cattle.platform.object.ObjectManager;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+
+import java.io.IOException;
+import java.util.Collections;
+
+import javax.inject.Inject;
+
+import com.netflix.config.DynamicStringProperty;
+
+public class StatsLinkHandler implements LinkHandler {
+
+    private static final DynamicStringProperty HOST_STATS_PATH = ArchaiusUtil.getString("host.stats.path");
+
+    HostApiService hostApiService;
+    ObjectManager objectManager;
+
+    @Override
+    public String[] getTypes() {
+        return new String[] { InstanceConstants.TYPE_CONTAINER, HostConstants.TYPE };
+    }
+
+    @Override
+    public boolean handles(String type, String id, String link, ApiRequest request) {
+        return StatsConstants.LINK_STATS.equals(link);
```

Comment:
I just moved the previous HostStatsLinkHandler to StatsLinkHandler. This is old code and we will remove this as soon as vince creates UI for this feature.